satpy/modifiers/angles.py

```diff
@@ -0,0 +1,266 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2021 Satpy developers
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+"""Utilties for getting various angles for a dataset.."""
+from __future__ import annotations
+
+import os
+import hashlib
+from datetime import datetime
+from functools import update_wrapper
+from glob import glob
+from typing import Callable, Any, Optional, Union
+import shutil
+
+import dask
+import dask.array as da
+import numpy as np
+import xarray as xr
+from satpy.utils import get_satpos, ignore_invalid_float_warnings
+from pyorbital.orbital import get_observer_look
+from pyorbital.astronomy import get_alt_az, sun_zenith_angle
+from pyresample.geometry import SwathDefinition, AreaDefinition
+
+import satpy
+
+PRGeometry = Union[SwathDefinition, AreaDefinition]
+
+# Arbitrary time used when computing sensor angles that is passed to
+# pyorbital's get_observer_look function.
+# The difference is on the order of 1e-10 at most as time changes so we force
+# it to a single time for easier caching. It is *only* used if caching.
+STATIC_EARTH_INERTIAL_DATETIME = datetime(2020, 1, 1, 0, 0, 0)
```

Comment:
```suggestion
STATIC_EARTH_INERTIAL_DATETIME = datetime(2000, 1, 1, 11, 58, 55, 816000)
```
Not that it really matters, but maybe use the astronomical standard epoch J2000? See https://en.wikipedia.org/wiki/Epoch_(astronomy)#Julian_dates_and_J2000